### DISCUSSION TITLE
Yosys frontend

### DISCUSSION THREAD

**Discussion by povik** [MikePopoloski/slang#886] [Show and tell]
Hello Michael and everyone else,

I have heard good things about slang, and have put together a prototype Yosys frontend. Let me drop the link here. 

https://github.com/povik/yosys-slang

There was some guesswork involved around slang APIs to get fast to something minimally functional, so feel free to call out whatever you would see that's wrong.

**Comment by MikePopoloski** [comment]
Very nice; I'll try to dig into this at some point and see if I have any suggestions for improvements.

**Comment by povik** [comment]
FWIW I opened a PR to get it on the sv-tests dashboard: https://github.com/chipsalliance/sv-tests/pull/5621

**Comment by udif** [comment]
Very nice!
Does it mean we now have a fully free SystemVerilog-compliant flow to check for combinatorial loops ?

**Reply by MikePopoloski** [reply]
In theory it could be turned into that, yes. The linked project is (I believe) just a proof of concept and doesn't support the full slang AST -- it just needs someone to go through and do the work to hook up everything.

**Reply by MikePopoloski** [reply]
Also there is a parallel effort to have slang act as a frontend to the LLVM CIRCT project, which presumably has a checker for combinational loops? (I don't actually know for sure).

**Reply by udif** [reply]
I also noticed the sv_lang scoreboard is broken. For some tests that has 10/10 for slang, have only 7/10 for slang_yosys when they produce the same output as slang. I think some tests are supposed to fail but the slang_yosys expected output is not defined correctly.

**Reply by povik** [reply]
> fully free SystemVerilog-compliant flow

Fully free, yes. SystemVerilog-compliant, not really. It's like Mike has said, there are unimplemented language features which make it unsuitable for production use just yet. Notable omissions would be the support of interfaces, or up-to-spec execution of initial blocks. (Though it might be argued the latter don't matter much for detection of combinational loops.) I would love to see it get full-featured but I can't justify the development effort right now.

**Reply by povik** [reply]
> For some tests that has 10/10 for slang, have only 7/10 for slang_yosys when they produce the same output as slang.

I don't have all the details right now, but it could be slang is run as a parser while yosys-slang is run as a synthesizer, so they are being tested for a different thing.
